### PR TITLE
 [App Course] As a waste picker, I want to be able to answer exercises - [Frontend] & [Backend]

### PR DESCRIPTION
**Pull request by Andazu**
## Description

[App Course]  "As a waste picker, I want to be able to review the exercises and get feedback from it"

## Changes

- Changed the review answer text into a button under the scrollview instead
- Made the confirm answer button change opacity when no answer has been selected
- Made it possible to press the text of the answer and not only the radio button
- Added background color to the scrollview dependant on correct or incorrect answers
- Added background color to the feedback text dependant on correct or incorrect answers
- Added icons to the feedback dependant on correct or incorrect answers
- Changed all tailwind values to percentage instead of pixels for responsiveness
- Updated the progressbar component to handle input parameters to change dimensions
- Updated api.js functions to correctly handle the data needed for exerciseScreen
- Removed unrelated remnant code which is not relevant
- Made the bottom portion of the exercise screen into its own component
- Made it possible give an exerciseId as an input parameter to the exerciseScreen on navigation (it has a random one by default if nothing is given, for test purposes)
- Imported functions from api.js to get an exercise, section and course
- Various visual fixes

## Related Issues

- In the future, the input parameters should be an entire exercise object and not only the Id. (The reason it is made this way is because the screen to navigate to the exerciseScreen has not been made)

## Checklist

- [X] Code has been tested locally and passes all relevant tests.
- [X] Documentation has been updated to reflect the changes, if applicable.
- [X] Code follows the established coding style and guidelines of the project.
- [X] All new and existing tests related to the changes have passed.
- [X] Any necessary dependencies or new packages have been properly documented.
- [X] Pull request title and description are clear and descriptive.
- [X] Reviewers have been assigned to the pull request.
- [X] Any potential security implications have been considered and addressed.
- [X] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)

Screenshot_1697709322

## Notes for Reviewers
There is currently a PNG in the exerciseInfo component. In the future, this entire PNG is going to be replaced with the component from the video streaming group.

When using the new progressbar component, it is possible to change its size to using this syntax:
```
<CustomProgressBar progress={%} width={%} height={%}></CustomProgressBar>

*REPLACE THE % WITH VALUE IN PERCENTAGE

```

### CODE REVIEW

File: eml/api/api.js
Comment (Jatewo) [inline eml/api/api.js:1]: Why are the quotes changed to single quotes? I believe we agreed on single quotes in the beginning of the project.

Comment (Jatewo) [reply eml/api/api.js:1]: This goes for all files

File: eml/components/exercise/ExerciseButtons.jsx [hunk 1]
```diff
@@ -48,14 +40,6 @@ export default function ExerciseButtons({ exerciseId, answers, sectionId, course
 
   const [button, setButton] = useState(true)
 
-  function handlePlaySound(tts) {
```
Comment (Jatewo) [inline eml/components/exercise/ExerciseButtons.jsx:51]: Why has this been deleted?

File: eml/components/exercise/ExerciseButtons.jsx [hunk 2]
```diff
@@ -7,14 +7,6 @@ import PropTypes from 'prop-types'
 import * as StorageService from '../../services/StorageService';
 import * as Speech from 'expo-speech'
 
-
-const voiceOvers = [
```
Comment (Jatewo) [inline eml/components/exercise/ExerciseButtons.jsx:11]: Why has this been deleted?

File: eml/screens/excercise/ExerciseScreen.jsx [hunk 3]
```diff
@@ -1,162 +1,139 @@
...
+import { useNavigation, useRoute } from "@react-navigation/native";
+import CustomProgressBar from "../../components/exercise/Progressbar";
+import { RadioButton } from "react-native-paper";
+import ExerciseInfo from "../../components/exercise/ExerciseInfo";
+import { Icon } from '@rneui/themed';
+import { SafeAreaView } from "react-native-safe-area-context";
+import { getExerciseByid, getSectionByid, getCourseByid } from '../../api/api';
 
-export default function ExerciseScreen() {
+export default function ExerciseScreen({ givenId = "65181a4f4c78b45368126ed7"}) {
```
Comment (Jatewo) [inline eml/screens/excercise/ExerciseScreen.jsx:13]: What is this givenId?

File: eml/api/userApi.js [hunk 4]
```diff
@@ -109,18 +115,24 @@ export const enrollInCourse = async (user_Id, course_Id) => {
...
       courseCompletion: res.data.course.isComplete,
-      sectionCompletion: res.data.course.sections.map(section => section.isComplete),
-      exerciseCompletion: res.data.course.sections.map(section => section.exercises.map(exercise => exercise.isComplete))
+      sectionCompletion: res.data.course.sections.map(
+        (section) => section.isComplete
+      ),
+      exerciseCompletion: res.data.course.sections.map((section) =>
+        section.exercises.map((exercise) => exercise.isComplete)
+      ),
     };
```
Comment (JackSMunn) [inline eml/api/userApi.js:136]: UpdateFirstname, updateLastname, UpdateEmail can be deleted. --> Ask @Mafusn if any questions

File: eml/screens/excercise/ExerciseScreen.jsx
Comment (Jatewo) [inline eml/screens/excercise/ExerciseScreen.jsx:1]: I don't understand why all this has been deleted - could you explain why? 😄

**Review by Jatewo — CHANGES_REQUESTED** [review]
### Great work
I have a few comments to this - some require changes, some might just require some explanation 👍

**Review by JackSMunn — DISMISSED** [review]
GJ! 😄 

There is only a single comment you should address from me.

File: eml/api/api.js [hunk 5]
```diff
@@ -1,20 +1,34 @@
...
+export const getSectionByid = async (sectionId) => {
+  const res = await axios.get(url + "/api/sections/" + sectionId);
+  return res.data;
+};
+
+export const getExerciseByid = async (exerciseId) => {
+  const res = await axios.get(url + "/api/exercises/" + exerciseId);
+  return res.data;
+};
+
```
Comment (Donhusum) [inline eml/api/api.js:25]: All three methods needs error handling

File: eml/api/api.js [hunk 6]
```diff
@@ -1,20 +1,34 @@
...
 // Get specific course
 
 export const getCourse = async (courseId) => {
   try {
-    const res = await axios.get(url + '/api/courses/' + courseId)
+    const res = await axios.get(url + "/api/courses/" + courseId);
```
Comment (Donhusum) [inline eml/api/api.js:30]: This function is doing the same as the new getCourseByid function. The code should be refactored so it uses the same one everywhere, and the redundant one should be deleted/out-commented

**Pull request closed without merging**